lib/oban/query.ex

```diff
@@ -70,11 +70,27 @@ defmodule Oban.Query do
   end
 
   @spec stage_scheduled_jobs(Config.t(), binary()) :: {integer(), nil}
-  def stage_scheduled_jobs(%Config{prefix: prefix, repo: repo, verbose: verbose}, queue) do
+  def stage_scheduled_jobs(%Config{} = config, queue) do
+    stage_scheduled_jobs(config, queue, fn queryable ->
+      where(queryable, [j], j.scheduled_at <= ^utc_now())
```

Comment:
Not sure if you had seen it, but I had a `@infinity_of_time DateTime.from_unix!(9_999_999_999)` in there for a while...

Not sure exactly, it didn't feel right. It would create a year 2248 problem.

Reply:
Maybe because it felt wasteful, to be precise. Even using the index, checking these timestamps for now reason.